### PR TITLE
feat: Improve rendering callbacks when Loading, Finished and Error

### PR DESCRIPTION
**Pull request by paulomeurerzup**
### Related Issues

Closes https://github.com/ZupIT/beagle/issues/746

### Description and Example

create ServerDrivenState.Success
replace ServerDrivenState.loading(Bool) with ServerDrivenState.Started and ServerDrivenState.Finished

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [ ] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt [hunk 1]
```diff
@@ -170,9 +196,22 @@ abstract class BeagleActivity : AppCompatActivity() {
     private fun handleLiveData(state: LiveData<ViewState>) {
         state.observe(this, Observer {
             when (it) {
-              is ViewState.Error -> onServerDrivenContainerStateChanged(ServerDrivenState.Error(it.throwable,it.retry))
-              is ViewState.Loading -> onServerDrivenContainerStateChanged(ServerDrivenState.Loading(it.value))
-              is ViewState.DoRender -> showScreen(it.screenId, it.component)
+              is ViewState.Error -> {
+                  onServerDrivenContainerStateChanged(ServerDrivenState.Error(it.throwable,it.retry))
```
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt:200]: add some space between `,` you can auto-indent using editor

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt [hunk 2]
```diff
@@ -170,9 +196,22 @@ abstract class BeagleActivity : AppCompatActivity() {
...
+              }
+              is ViewState.Loading -> {
+                  onServerDrivenContainerStateChanged(ServerDrivenState.Loading(it.value))
+
+                  if(it.value) {
```
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt:205]: the problem using this approach when some error happens will emit the `Finished` and  `Error` and this is not a good idea.
The best approach is only emitting the `finished` if don't has an error.

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleView.kt [hunk 3]
```diff
@@ -80,14 +92,23 @@ internal class BeagleView(
         } else {
             BeagleViewState.LoadFinished
         }
+
+        val serverState = if (isLoading) {
```
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleView.kt:96]: the same comment above.

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt [hunk 4]
```diff
@@ -41,10 +42,35 @@ import br.com.zup.beagle.core.ServerDrivenComponent
...
+
     class FormError(throwable: Throwable, retry: BeagleRetry) : Error(throwable, retry)
     class WebViewError(throwable: Throwable, retry: BeagleRetry) : Error(throwable, retry)
+    @Deprecated(DEPRECATED_STATE_LOADING)
     data class Loading(val loading: Boolean) : ServerDrivenState()
+
+    /**
+     * indicates that a server-driven component fetch has begun
+     */
+    open class Started: ServerDrivenState()
```
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt:54]: there is some reason to use `open class`? what do you think to convert to `object`
```suggestion
    object Started: ServerDrivenState()
```

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt (hunk 2 above)
Comment (paulomeurerzup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt:205]: I think the same, but @tuliopereirazup and iOS team defined that `finished` always will be called.

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleView.kt (hunk 3 above)
Comment (paulomeurerzup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleView.kt:96]: I think the same, but @tuliopereirazup and iOS team defined that `finished` always will be called.

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt (hunk 4 above)
Comment (paulomeurerzup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt]: done

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt (hunk 1 above)
Comment (paulomeurerzup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/view/BeagleActivity.kt]: done

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**